document immutable field resolution limitation in the same class (#587)
MicroEJ/docs: KernelDeveloperGuide/applicationLink.rst

- its Architecture version is compatible [#compatible_def]_ with the Architecture version on which the ``.fso`` file has been produced,
- it provides the required APIs according to the JVM specification [1]_.

> A current limitation is that if the Sandboxed Application declares an immutable object, SOAR Optimizer will resolve fields in the same class, not the whole class hierarchy.

Reviewer: ```suggestion
A current limitation is that if the Sandboxed Application declares an immutable object, SOAR Optimizer will resolve fields within the same class rather than considering the entire class hierarchy.
```
